eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -187,7 +187,7 @@ private function internalPublishUrlAliasForLocation(
         while (true) {
             $newText = '';
             if ($locationId != self::CONTENT_REPOSITORY_ROOT_LOCATION_ID) {
-                $newText = $name . ($uniqueCounter > 1 ? $uniqueCounter : '');
+                $newText = $name . ($name == '' || $uniqueCounter > 1 ? $uniqueCounter : '');
```

Comment:
How would this behave in case there was more than one names which results in an empty string? My assumption is that `$uniqueCounter` still would be 1, which worries me a bit.

Reply:
@alongosz Duplicates can happen if there are pre-existing db entries. The counter will be increased by 1 then, see where it deals with UniqueConstraintViolationException 30 lines below, and at the end of the while loop.